Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -0,0 +1,194 @@
+//
+//  OCTAudioEngine.m
+//  objcTox
+//
+//  Created by Chuong Vu on 5/24/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "OCTAudioEngine.h"
+@import AVFoundation;
+#define kInputBus 1
+#define kOutputBus 0
+
+@interface OCTAudioEngine ()
+{
+    AUGraph processingGraph;
+
+    AUNode ioNode;
+    AudioUnit ioUnit;
+    AudioComponentDescription ioUnitDescription;
+}
+
+@property (nonatomic, assign) double sampleRate;
+
+@end
+
+@implementation OCTAudioEngine
+
+#pragma mark - LifeCycle
+-(instancetype)init
+{
+    self = [super init];
+    if (!self)
+        return nil;
+
+    ioUnitDescription.componentType = kAudioUnitType_Output;
+    ioUnitDescription.componentSubType = kAudioUnitSubType_VoiceProcessingIO;
+    ioUnitDescription.componentManufacturer = kAudioUnitManufacturer_Apple;
+    ioUnitDescription.componentFlags = 0;
+    ioUnitDescription.componentFlagsMask = 0;
+
+    NewAUGraph(&processingGraph);
+    CheckStatus(NewAUGraph(&processingGraph),
+                @"Failed to create AUGraph");
+
+    CheckStatus(AUGraphAddNode(processingGraph,
+                   &ioUnitDescription,
+                   &ioNode),
+                @"Failed to add ioNode");
+
+    CheckStatus(AUGraphOpen(processingGraph),
+                @"Failed to open graph");
+
+    AUGraphNodeInfo(processingGraph, ioNode, NULL, &ioUnit);
+
+    return self;
+}
+
+#pragma mark - Audio Controls
+-(BOOL)startAudioFlow:(NSError **)error
+{
+    bool success = [self startAudioSession:error];
+    if (!success) {
+        return success;
+    }
```

Comment:
Code like this all over the file can be refactored to
```
if (! [self startAudioSession:error) {
    return NO;
}
```

which looks much nicer to me. There is no need in success variable and code is more compact.